crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -605,3 +608,50 @@ impl StateChanges {
         self.timeline.insert(room_id.to_owned(), timeline);
     }
 }
+
+/// Configuration for the state store and, when `encryption` is enabled, for the
+/// crypto store.
+///
+/// # Example
+///
+/// ```
+/// # use matrix_sdk_base::store::StoreConfig;
+///
+/// let store_config = StoreConfig::new();
+/// ```
+#[derive(Default)]
+pub struct StoreConfig {
+    #[cfg(feature = "encryption")]
+    pub(crate) crypto_store: Option<Box<dyn CryptoStore>>,
+    pub(crate) state_store: Option<Box<dyn StateStore>>,
+}
+
+#[cfg(not(tarpaulin_include))]
+impl std::fmt::Debug for StoreConfig {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> StdResult<(), std::fmt::Error> {
+        fmt.debug_struct("StoreConfig").finish()
+    }
+}
+
+impl StoreConfig {
+    /// Create a new default `StoreConfig`.
+    #[must_use]
+    pub fn new() -> Self {
+        Default::default()
+    }
+
```

Comment:
how about having `new_with_state_store` and (if enabled) `new_with_state_and_crypto_store`? That would also make the usage examples less verbose.

Reply:
```suggestion
    /// Create a new store config wrapping the given state store
    pub fn new_with_state_store(state_store: Box<dyn StateStore>) -> Self {
        StoreConfig {
            state_store,
            ..Default::default()
        }
    }
    
    /// Create a new store config wrapping the given state and crypto store
    #[cfg(feature = "encryption")]
    pub fn new_with_state_and_crypto_store(state_store: Box<dyn StateStore>, crypto_store: Box<dyn CryptoStore>) -> Self {
        StoreConfig {
            state_store,
            crypto_store
            ..Default::default()
        }
    }
```

Reply:
Should I drop the two other methods then? They seem useless now.

Reply:
Ah nevermind, we need them unless we change the API of `ClientConfig` too.